src/Common/ThreadStatus.cpp

```diff
@@ -47,28 +56,42 @@ struct ThreadStack
 {
     ThreadStack()
     {
-        data = aligned_alloc(getPageSize(), getSize());
+        auto page_size = getPageSize();
+        data = aligned_alloc(page_size, getSize());
         if (!data)
             throw ErrnoException(ErrorCodes::CANNOT_ALLOCATE_MEMORY, "Cannot allocate ThreadStack");
 
-        try
-        {
-            /// Since the stack grows downward, we need to protect the first page
-            memoryGuardInstall(data, getPageSize());
-        }
-        catch (...)
+        if (guardPagesEnabled())
         {
-            free(data);
-            throw;
+            try
+            {
+                /// Since the stack grows downward, we need to protect the first page
+                memoryGuardInstall(data, page_size);
+            }
+            catch (...)
+            {
+                free(data);
+                throw;
+            }
         }
     }
     ~ThreadStack()
     {
-        memoryGuardRemove(data, getPageSize());
+        if (guardPagesEnabled())
+            memoryGuardRemove(data, getPageSize());
+
         free(data);
     }
 
-    static size_t getSize() { return std::max<size_t>(UNWIND_MINSIGSTKSZ, MINSIGSTKSZ); }
+    static size_t getSize()
+    {
+        auto size = std::max<size_t>(UNWIND_MINSIGSTKSZ, MINSIGSTKSZ);
+
+        if (guardPagesEnabled())
```

Comment:
```suggestion
        if constexpr (guardPagesEnabled())
```

Reply:
But actually, why do we need this? Previously we were using the same allocation. But even if you want extra page for guard, you need to add a `getPageSize()`

Reply:
For me it looks inconsistent when we use additional page in FiberStack and do not used it in altstack. So I just added additional page when guard is enabled.

Reply:
Ok, I guess it should not be a big deal, for 10k threads only 40MiBs